### DISCUSSION TITLE
How to snapshot on special iOS version?

### DISCUSSION THREAD

**Discussion by davidhu001** [pointfreeco/swift-snapshot-testing#825] [General]
SnapshotTesting uses GitHub issues for bugs. For more general discussion and help, please use [GitHub Discussions](https://github.com/pointfreeco/swift-snapshot-testing/discussions).

Hi @stephencelis @mdiep @gohanlon @ChristopherRogers @heckj 
I want to snapshot of a special iOS device and a special iOS OS version, but I find I only can do is set the special iOS device, but I can't set the special iOS OS version.

`        assertSnapshots(
                of: hostingController,
                as: [
                  "ipad": .image(on: .iPadPro12_9),
                  "iphoneSe": .image(on: .iPhoneSe),
                  "iphone8": .image(on: .iPhone8),
                  "iphoneMax": .image(on: .iPhoneXsMax),
                ])`
I got above this from the Demo, But I haven't found any OK solution for adding the special OS version.

Could you give me some advice so that I can set up the special OS version?

Thank you very much.

**Comment by mbrandonw** [comment]
Hi @davidhu001, this is not an issue with the library and so I am converting it to a discussion.

**Reply by davidhu001** [reply]
thank you @mbrandonw

**Comment by fespinoza** [comment]
I don't think you can specify the iOS version at runtime, like you decide device.

One thing you can do is to have an Xcode Test Plan, where you include the specific tests that have references to an older iOS version. Then on CI run only that test plan in the given iOS version. (This means you will need several actions)

Or another option, in Xcode Cloud, is to run the tests in multiple environments and on runtime check if the test is meant to be run in a particular iOS version or skip it otherwise.
